### PR TITLE
enh(web): refactor loginSession to use to prepare password expiration

### PR DESCRIPTION
**Pull request by kduret**
## Description

refactor loginSession to use to prepare password expiration

**Fixes** MON-12289

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

### CODE REVIEW

File: config/packages/Centreon.yaml [hunk 1]
```diff
@@ -322,6 +322,10 @@ services:
     Centreon\Domain\Menu\Interfaces\MenuRepositoryInterface:
         class: Centreon\Infrastructure\Menu\MenuRepositoryRDB
 
+    Core\Application\Security\UseCase\LoginSession\LoginSession:
+        arguments: [ "%redirect_default_page%" ]
+        public: true
+
     Centreon\Domain\Authentication\UseCase\Authenticate:
```
Comment (kduret) [inline config/packages/Centreon.yaml:329]: remove this

File: config/routes/Centreon/security/authentication.yaml [hunk 2]
```diff
@@ -4,9 +4,9 @@ centreon_security_authentication_find_providers_configurations:
     controller: 'Centreon\Application\Controller\AuthenticationController::findProvidersConfigurations'
 
 centreon_security_authentication_login:
-    methods: GET|POST
+    methods: POST
     path: /authentication/providers/configurations/{providerConfigurationName}
```
Comment (kduret) [inline config/routes/Centreon/security/authentication.yaml:8]: check requirement for `providerConfigurationName`

File: src/Core/Application/Common/UseCase/AbstractPresenter.php [hunk 3]
```diff
@@ -45,11 +44,20 @@ public function __construct(protected PresenterFormatterInterface $presenterForm
...
      * @inheritDoc
      */
-    public function show(): Response
+    public function present(mixed $data): void
     {
         if ($this->getResponseStatus() !== null) {
             $this->presenterFormatter->present($this->getResponseStatus());
+        } else {
+            $this->presenterFormatter->present($data);
         }
```
Comment (kduret) [inline src/Core/Application/Common/UseCase/AbstractPresenter.php:53]:
```suggestion
        $this->presenterFormatter->present($data);
```

File: src/Core/Application/Common/UseCase/CreatedResponse.php [hunk 4]
```diff
@@ -23,8 +23,6 @@
 
 namespace Core\Application\Common\UseCase;
 
-use Core\Application\Common\UseCase\ResponseStatusInterface;
```
Comment (kduret) [inline src/Core/Application/Common/UseCase/CreatedResponse.php:26]: retrieve use

File: src/Core/Application/Common/UseCase/ErrorResponse.php [hunk 5]
```diff
@@ -23,8 +23,6 @@
 
 namespace Core\Application\Common\UseCase;
 
-use Core\Application\Common\UseCase\ResponseStatusInterface;
```
Comment (kduret) [inline src/Core/Application/Common/UseCase/ErrorResponse.php:26]: retrieve use

File: src/Core/Application/Common/UseCase/NoContentResponse.php [hunk 6]
```diff
@@ -23,8 +23,6 @@
 
 namespace Core\Application\Common\UseCase;
 
-use Core\Application\Common\UseCase\ResponseStatusInterface;
```
Comment (kduret) [inline src/Core/Application/Common/UseCase/NoContentResponse.php:26]: retrieve use

File: src/Core/Application/Common/UseCase/NotFoundResponse.php [hunk 7]
```diff
@@ -23,8 +23,6 @@
 
 namespace Core\Application\Common\UseCase;
 
-use Core\Application\Common\UseCase\ResponseStatusInterface;
```
Comment (kduret) [inline src/Core/Application/Common/UseCase/NotFoundResponse.php:26]: retrieve use

File: src/Core/Application/Security/UseCase/LoginSession/LoginSession.php [hunk 8]
```diff
@@ -102,76 +59,85 @@ class Authenticate
...
+                        ),
+                        $request->clientIp,
+                    );
+                }
+            }
+        } catch (PasswordExpiredException $e) {
+            $response = new PasswordExpiredResponse($e->getMessage());
+            $response->setBody([
+                'password_is_expired' => true,
+            ]);
```
Comment (kduret) [inline src/Core/Application/Security/UseCase/LoginSession/LoginSession.php:129]: why not use dto ?

File: src/Security/Domain/Authentication/Model/LocalProvider.php [hunk 9]
```diff
@@ -132,16 +130,36 @@ function () use ($auth) {
...
             if ($auth->userInfos !== null) {
                 $this->contactId = (int) $auth->userInfos['contact_id'];
                 $this->setLegacySession(new \Centreon($auth->userInfos));
             }
-            $this->isAuthenticated = true;
             $this->info('[LOCAL PROVIDER] authentication succeed');
         } else {
-            $this->isAuthenticated = false;
-            $this->info('[LOCAL PROVIDER] authentication failed');
+            $this->critical(
```
Comment (kduret) [inline src/Security/Domain/Authentication/Model/LocalProvider.php:155]:
```suggestion
            $this->info(
```

File: config/routes/Centreon/security/authentication.yaml (hunk 2 above)
Comment (kduret) [reply config/routes/Centreon/security/authentication.yaml:8]: regarding documentation, requirements seems only useful in specific case like route collision : https://symfony.com/doc/current/routing.html#parameters-validation
Then, I don't add it here

File: src/Core/Application/Security/UseCase/LoginSession/LoginSession.php (hunk 8 above)
Comment (kduret) [reply src/Core/Application/Security/UseCase/LoginSession/LoginSession.php:129]: Cannot use DTO here cause AbstractPresenter cannot handle it

**Review by jeremyjaouen — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**